server/src/test/java/com/tencent/rss/server/HealthCheckTest.java

```diff
@@ -0,0 +1,87 @@
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.tencent.rss.server;
+
+import org.junit.Test;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class HealthCheckTest {
+
+  @Test
+  public void constructorTest() {
+    ShuffleServerConf conf = new ShuffleServerConf();
+    assertConf(conf);
+    conf.setString(ShuffleServerConf.RSS_HEALTH_CHECKER_CLASS_NAMES, "");
+    assertConf(conf);
+    conf.setString(ShuffleServerConf.RSS_HEALTH_CHECKER_CLASS_NAMES, "com.tencent.rss.server.StorageChecker");
+    conf.set(ShuffleServerConf.RSS_STORAGE_BASE_PATH, "s1");
+    conf.set(ShuffleServerConf.RSS_HEALTH_MIN_STORAGE_PERCENTAGE, -1.0);
+    assertConf(conf);
+    conf.set(ShuffleServerConf.RSS_HEALTH_MIN_STORAGE_PERCENTAGE, 102.0);
+    assertConf(conf);
+    conf.set(ShuffleServerConf.RSS_HEALTH_MIN_STORAGE_PERCENTAGE, 1.0);
+    conf.set(ShuffleServerConf.RSS_HEALTH_CHECK_INTERVAL, -1L);
+    assertConf(conf);
+    conf.set(ShuffleServerConf.RSS_HEALTH_CHECK_INTERVAL, 1L);
+    conf.set(ShuffleServerConf.RSS_HEALTH_STORAGE_MAX_USAGE_PERCENTAGE, -1.0);
+    assertConf(conf);
+    conf.set(ShuffleServerConf.RSS_HEALTH_STORAGE_MAX_USAGE_PERCENTAGE, 101.0);
+    assertConf(conf);
+    conf.set(ShuffleServerConf.RSS_HEALTH_STORAGE_MAX_USAGE_PERCENTAGE, 1.0);
+    conf.set(ShuffleServerConf.RSS_HEALTH_STORAGE_RECOVERY_USAGE_PERCENTAGE, -1.0);
+    assertConf(conf);
+    conf.set(ShuffleServerConf.RSS_HEALTH_STORAGE_RECOVERY_USAGE_PERCENTAGE, 101.0);
+    assertConf(conf);
+    conf.set(ShuffleServerConf.RSS_HEALTH_STORAGE_RECOVERY_USAGE_PERCENTAGE, 1.0);
+    new HealthCheck(new AtomicBoolean(), conf);
+  }
+
+  @Test
+  public void checkTest() {
+    AtomicBoolean healthy = new AtomicBoolean(false);
+    ShuffleServerConf conf = new ShuffleServerConf();
+    conf.setString(ShuffleServerConf.RSS_HEALTH_CHECKER_CLASS_NAMES, MockChecker2.class.getCanonicalName());
+    HealthCheck checker = new HealthCheck(healthy, conf);
+    checker.check();
+    assertTrue(healthy.get());
+    conf.setString(ShuffleServerConf.RSS_HEALTH_CHECKER_CLASS_NAMES, MockChecker1.class.getCanonicalName());
+    checker = new HealthCheck(healthy, conf);
+    checker.check();
+    assertFalse(healthy.get());
+    conf.setString(ShuffleServerConf.RSS_HEALTH_CHECKER_CLASS_NAMES,
+        MockChecker1.class.getCanonicalName() + "," + MockChecker2.class.getCanonicalName());
+    checker = new HealthCheck(healthy, conf);
+    checker.check();
+    assertFalse(healthy.get());
+  }
+
+  private void assertConf(ShuffleServerConf conf) {
+    boolean isThrown;
+    isThrown = false;
+    try {
+      new HealthCheck(new AtomicBoolean(), conf);
+    } catch (IllegalArgumentException e) {
+      isThrown = true;
+    }
+    assertTrue(isThrown);
+  }
+}
```

Comment:
empty line

Reply:
empty line is a must since the warning is here.

Reply:
We cound fix it since github give us a kind warning. Spark and most Java code also comply with the rule, this is a non-mandatory github warining, I will fix it in our checksytle to make it mandatory.

Reply:
Spark don't follow this rule........

Reply:
I misunderstand the rule that you wanna express.I will fix .